site/src/modules/resources/AppLink/AppLink.tsx

```diff
@@ -149,12 +149,22 @@ export const AppLink: FC<AppLinkProps> = ({ app, workspace, agent }) => {
 							setFetchingSessionToken(false);
 						}
 						window.location.href = url;
-					} else {
-						window.open(
-							href,
-							Language.appTitle(appDisplayName, generateRandomString(12)),
-							"width=900,height=600",
-						);
+						return;
+					}
+
+					switch (app.open_in) {
+						case "slim-window": {
+							window.open(
+								href,
+								Language.appTitle(appDisplayName, generateRandomString(12)),
+								"width=900,height=600",
+							);
+							return;
+						}
+						case "tab": {
```

Comment:
Should this be the `default` case or are we guaranteed `open_in` will always be set and only contain one of these two values?

Reply:
We ensure it will always be set and contain one of these two values.